### ISSUE TITLE
Unable to create a class under windows (Path.sep replace issue)

### ISSUE THREAD

**Issue by shrayasr** [jchannon/csharpextensions#4]
I'm unable to create a class when using this extension in Windows.

Looks like there is a path separator issue. When I try to create a **new C# class**, the following error is thrown at the console

```
SyntaxError: Invalid regular expression: /\/: \ at end of pattern
    at RegExp (native)
    at vscode.window.showInputBox.then (C:\Users\shrayasr\.vscode\extensions\jchannon.csharpextensions-1.0.1\out\src\extension.js:51:39)
    at Object.m [as _notify] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:5002)
    at Object.enter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:8580)
    at _run (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:10408)
    at _completed (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:9849)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:16511
    at done (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:11709)
    at Object.m [as _notify] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:5144)
    at Object.enter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:8580)
    at _run (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:10408)
    at _completed (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:6:9849)
    at e.resolveOk (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:8:4013)
    at p (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:8:2495)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

Line 51 of the JS file is this:

```javascript
namespace = namespace.replace(new RegExp(path.sep, 'g'), '.');
```

I tried out something simple in a REPL and ended up with the same issue

```javascript
λ node                                                                 
> var path = require("path");                                          
undefined                                                              
> path.sep                                                             
'\\'                                                                   
> "myproject\\controllers".replace(new RegExp(path.sep, "g"), ".");    
SyntaxError: Invalid regular expression: /\/: \ at end of pattern      
    at new RegExp (native)                                             
    at repl:1:34                                                       
    at REPLServer.defaultEval (repl.js:248:27)                         
    at bound (domain.js:280:14)                                        
    at REPLServer.runBound [as eval] (domain.js:293:12)                
    at REPLServer.<anonymous> (repl.js:412:12)                         
    at emitOne (events.js:82:20)                                       
    at REPLServer.emit (events.js:169:7)                               
    at REPLServer.Interface._onLine (readline.js:210:10)               
    at REPLServer.Interface._line (readline.js:549:8)                  
```

It works under the nix platforms because `path.sep` will be `/` and this works fine 

```javascript
> "myproject/controllers".replace(new RegExp("/", "g"), ".");
'myproject.controllers'
```

I will try to put out a PR for this today

**Comment by jchannon** [comment]
Time to get @Yantrio mystery magical regex machine out ;)

**Comment by shrayasr** [comment]
One way to solve this is to not use `path.sep` but just directly use `/` as a separator. Works on windows too, I think? 

(read: hack hack hack)

**Comment by jchannon** [comment]
I have this regex which works in a debugger but blows up in node for some reason.  Won't have any time to fix today 😢 

```
> '/Users/jc/projects/nancy/src/fred/poo/'.replace(new RegExp('\/|\\', 'g'), '.');
SyntaxError: Invalid regular expression: //|\/: \ at end of pattern
```

**Comment by jchannon** [comment]
I think I have it

` 'asd\\asd\\asd\\asd'.replace(new RegExp('\/|\\\\', 'g'), '.');`

`'/Users/jc/projects/nancy/src/fred/poo/'.replace(new RegExp('\/|\\\\', 'g'), '.');`

**Comment by jchannon** [comment]
Bugger fails on windows

**Comment by shrayasr** [comment]
```
SUCH REGEX 
    MUCH PLATFORMS
  VERY VARIATIONS
AMAZE
```

:disappointed:

**Comment by shrayasr** [comment]
Looking into this, I also realize that this isn't the least of our problems under Windows. 

We also have this line:

```javascript
var newroot = parentdir.substr(parentdir.lastIndexOf('/') + 1);
```

In my case, I have `parentdir` set to `"c:\\Users\\shrayasr\\work\\code\\ixm\\src\\ixm"`. Now consider a unix env, `parentdir` would be `/users/shrayasr/work/code/ixm/src/ixm`. `newroot` is set differently on both platforms because of the char we use to substring

REPL output: 

```javascript
> var parentdir = "/users/shrayasr/work/code/ixm/src/ixm"
> var newroot = parentdir.substr(parentdir.lastIndexOf('/') + 1);
> newroot
'ixm'
> var parentdir = "c:\\Users\\shrayasr\\work\\code\\ixm\\src\\ixm"
> var newroot = parentdir.substr(parentdir.lastIndexOf('/') + 1);
> newroot
'c:\\Users\\shrayasr\\work\\code\\ixm\\src\\ixm'
```

with `newroot` being set this way, it doesn't matter if we fix anything below it, the namespace will not be right. Manually continuing leaves me with this namespace: 

```
"c:.Users.shrayasr.work.code.ixm.src.ixm.Controllers"
```

hehe. Thoughts?

**Comment by jchannon** [comment]
yeah i spotted that too :)

**Comment by shrayasr** [comment]
Taken an initial whack in the PR. Do give it a glance. Works for me now :+1:

**Comment by jchannon** [comment]
Closed via #5
